temporian/core/operators/filter.py

```diff
@@ -0,0 +1,118 @@
+# limitations under the License.
+
+"""Filter operator."""
+from temporian.core import operator_lib
+from temporian.core.data.dtype import BOOLEAN
+from temporian.core.data.feature import Feature
+from temporian.core.data.event import Event
+from temporian.core.data.sampling import Sampling
+from temporian.core.operators.base import Operator
+from temporian.proto import core_pb2 as pb
+
+
+class FilterOperator(Operator):
+    """Filter operator."""
+
+    def __init__(self, event: Event, condition: Event):
+        super().__init__()
+
+        # check that condition is a single feature
+        if len(condition.features) != 1:
+            raise ValueError(
+                "Condition must be a single feature. Got"
+                f" {len(condition.features)} instead."
+            )
+
+        # check that condition is a boolean feature
+        if condition.features[0].dtype != BOOLEAN:
+            raise ValueError(
+                "Condition must be a boolean feature. Got"
+                f" {condition.features[0].dtype} instead."
+            )
+
+        # check both events have same sampling
+        if event.sampling != condition.sampling:
+            raise ValueError(
+                "Event and condition must have the same sampling. Got"
+                f" {event.sampling} and {condition.sampling} instead."
+            )
+
+        # inputs
+        self.add_input("event", event)
+        self.add_input("condition", condition)
+
+        output_sampling = Sampling(index=event.sampling.index, creator=self)
+
+        self.condition_name = condition.features[0].name
+
+        # outputs
+        output_features = [  # pylint: disable=g-complex-comprehension
+            Feature(
+                name=self.feature_name(f),
+                dtype=f.dtype,
+                sampling=output_sampling,
+                creator=self,
+            )
+            for f in event.features
+        ]
+
+        self.add_output(
+            "event",
+            Event(
+                features=output_features,
+                sampling=output_sampling,
+                creator=self,
+            ),
+        )
+
+        self.check()
+
+    @classmethod
+    def build_op_definition(cls) -> pb.OperatorDef:
+        return pb.OperatorDef(
+            key="FILTER",
+            attributes=[],
+            inputs=[
+                pb.OperatorDef.Input(key="event"),
+                pb.OperatorDef.Input(key="condition"),
+            ],
+            outputs=[pb.OperatorDef.Output(key="event")],
+        )
+
+    def feature_name(self, feature: Feature) -> str:
+        return f"{feature.name}_filtered_by_{self.condition_name}"
+
+
+operator_lib.register_operator(FilterOperator)
+
+
+# Added `_event` to the function name to avoid name collision with the
+# `filter` function in the `builtins` module.
+def filter_event(
+    event: Event,
+    condition: Event,
+) -> Event:
+    """Filter operator.
```

Comment:
Nit: Should this be conjugated to the 3rd person?

For example, ""Filters out events for which the condition is false."""

https://github.com/google/temporian/pull/82/files#diff-ad7254a7a0a912467b0cc68a46f2f6f5560e26c992d14372c4bf7c5b85275795